### ISSUE TITLE
Task#onSuccess swallows exceptions in the callback

### ISSUE THREAD

**Issue by caneleex** [DV8FromTheWorld/JDA#1513]
[download]: https://bintray.com/dv8fromtheworld/maven/JDA/_latestVersion
[guild]: https://discord.gg/0hMr4ce0tIk3pSjp
[stack overflow]: https://stackoverflow.com/questions/tagged/java

## General Troubleshooting

  
- [x] I have checked for similar issues.
- [x] I have updated to the [latest JDA version][download].
- [x] I have checked the branches or the maintainers' PRs for upcoming bug fixes.

### Expected Behavior

Exceptions in the onSuccess callback should be logged

### Code Example or Reproduction Steps

```java
guild.loadMembers().onSuccess(members -> {
    throw new NullPointerException(); // the exception is not logged
}).onError(throwable -> logger.error(throwable.getMessage())); // neither it's logged/caught by the onError callback which is a bit unrelated
		
guild.loadMembers(member -> {
    throw new NullPointerException(); // the exception is logged
});

guild.retrieveMemberById(394607709741252621L).queue(member -> {
    throw new NullPointerException(); // the exception is logged
});
```

### Code for JDABuilder or DefaultShardManagerBuilder Used

N/A

### Exception or Error

N/A

**Comment by Alontrle** [comment]
I am having this exact issue with the addRoleToMember call. I supply a consumer for the action and the exception. Neither are triggered. The API works 10% of the time and fails 90% of the time without hitting any of my exceptions or logs.

I have provided the code with the error in the pastebin. I have also provided a full log that shows that the code before the call is being called successfully. It also provides all the JDA debug information if that can be used to further replicate the issue.

The code in question: https://pastebin.com/dL3bByMR
The full log: https://drive.google.com/file/d/1foa_2juV2TopXQGyxTA9V1Wg14zsqLAr/view?usp=sharing

**Comment by MinnDevelopment** [comment]
I can't read that google drive link but if an error is logged then its clearly not the same issue. `addRoleToMember` also doesn't return a `Task`. To handle exceptions use the failure callback of queue. See also [Troubleshooting](https://github.com/DV8FromTheWorld/JDA/wiki/19\)-Troubleshooting#restaction-queue-returned-failure).

**Comment by Alontrle** [comment]
> I can't read that google drive link but if an error is logged then its clearly not the same issue. `addRoleToMember` also doesn't return a `Task`. To handle exceptions use the failure callback of queue. See also [Troubleshooting](https://github.com/DV8FromTheWorld/JDA/wiki/19)-Troubleshooting#restaction-queue-returned-failure).

My bad, I'll open my own issue report. I do use a failure callback, it is not being called.

**Comment by MinnDevelopment** [comment]
This is resolved with [4.2.0_243](https://ci.dv8tion.net/job/JDA/243/). This issue can stay open until 4.2.1 is released.
